packages/data-broker/src/__tests__/services/dhis/pullers/EventsPuller.test.js

```diff
@@ -86,13 +95,107 @@ describe('EventsPuller', () => {
     it('translates data source element to DHIS element codes if required', () =>
       assertEventAnalyticsApiWasInvokedCorrectly({
         dataSources: [DATA_GROUPS.POP01_GROUP],
-        options: { dataElementCodes: ['POP01', 'DIF01'] },
+        options: { dataElementCodes: ['POP01', 'DIF01'], organisationUnitCodes: ['DL_FACILITY_A'] },
         invocationArgs: expect.objectContaining({ dataElementCodes: ['POP01', 'DIF01_DHIS'] }),
       }));
   });
 
   describe('data building', () => {
+    describe('pulling events', () => {
+      it('can pull events', async () => {
+        const getEventAnalyticsResponse = {
+          headers: [{ name: 'POP01', column: 'Population 1', valueType: 'NUMBER' }],
+          metaData: {
+            items: {
+              POP01: { name: 'Population 1' },
+            },
+            dimensions: { POP01: [] },
+          },
+          rows: [[5], [4]],
+        };
+
+        dhisApi = createMockDhisApi({ getEventAnalyticsResponse });
+        stubGetDhisApi(dhisApi);
+        const dataElementCodes = ['POP01', 'POP02'];
+        const organisationUnitCodes = ['DL_FACILITY_A'];
+
+        const events = await eventsPuller.pull([dhisApi], [DATA_GROUPS.POP01_GROUP], {
+          dataElementCodes,
+          organisationUnitCodes,
+        });
+
+        expect(events).toEqual([{ dataValues: { POP01: 5 } }, { dataValues: { POP01: 4 } }]);
+      });
+
+      it('can pull data for just the requested tracked entity instance', async () => {
+        // Return data for all children of the parent org unit
+        const getEventAnalyticsStub = ({ organisationUnitCodes }) => {
+          if (organisationUnitCodes[0] !== 'DL_FACILITY_A' || organisationUnitCodes.length !== 1) {
```

Comment:
I think I get it but not 100%, is this because its called twice, once for org units and the second time for tracked entities? If so can you add a comment

Reply:
Honestly it's actually not needed, since it does only get called once. I just sort of wanted to make it a bit explicit that it's returning data for the parent (`DL_FACILITY_A`) even though we're querying for the child (`DL_HOUSEHOLD_1`). But it's a bit confusing fair point, I'll add a comment to make it more clear 👍